pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -240,51 +204,6 @@ def _make_step_fn(self, split_batch: Any, batch_idx: int, opt_idx: int, hiddens:
         """Build the step function that runs the `training_step` and processes its output."""
         return partial(self._training_step, split_batch, batch_idx, opt_idx, hiddens)
 
-    def _make_zero_grad_fn(self, batch_idx: int, opt_idx: int, optimizer: Optimizer) -> Optional[Callable[[], None]]:
```

Comment:
reviewer: this removed block is all specific to automatic optimization and gets moved to the OptimizerLoop